apps/mobile/src/components/DrawerNavigation/Sidebar.tsx

```diff
@@ -0,0 +1,53 @@
+import type { FC } from 'react';
+import { View } from 'react-native';
+import type { DrawerContentComponentProps } from '@react-navigation/drawer';
+import DashboardNavigator from '@walless/app/features/DashboardLayout/Navigator';
+import {
+	useSafeAreaInsets,
+	useSnapshot,
+	useWidgets,
+} from '@walless/app/utils/hooks';
+import { appState, mockWidgets, widgetActions } from '@walless/engine';
+import type { WidgetDocument } from '@walless/store';
+import { appActions } from 'state/app';
+
+export const sidebarWidth = 64;
+
+const IS_HARDCODED = true;
+
+export const Sidebar: FC<DrawerContentComponentProps> = ({ navigation }) => {
+	const { profile, activeWidgetId } = useSnapshot(appState);
+	const widgets = useWidgets();
+
+	const handleExtensionPress = (item: WidgetDocument) => {
+		const id = item._id || 'Explore';
+		appActions.setActiveWidgetRoute(id);
+		navigation.navigate(id);
+	};
+
+	const handleRemoveWidget = async (widget: WidgetDocument) => {
+		await widgetActions.removeWidget(widget);
+		await navigation.navigate('Explore');
+	};
+
+	const getRouteActive = (item: WidgetDocument) => {
+		return activeWidgetId === (item._id || 'Explore');
+	};
+
+	const { top } = useSafeAreaInsets();
```

Comment:
hooks should be on top, together in a single block -- move this line above next to `useWidgets()`